x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -631,6 +615,10 @@ static Set<String> fieldNames(LogicalPlan parsed, Set<String> enrichPolicyMatchF
         });
         // Add JOIN ON column references afterward to avoid Alias removal
         references.addAll(keepJoinReferences);
+        // If any JOIN commands need wildcard field-caps calls, persist the index names
```

Comment:
Is this comment ok for every JOIN command we'll have, or is something specific to `lookup join`.
Either way, I suggest being specific.

Reply:
This is true as long as the LOOKUP JOIN command does not have a `WITH` clause, which is true at the moment for all joins, but will not be true at some future point. We can only fix this once that syntax exists. Without `WITH` we imply `*`.